legate/core/solver.py

```diff
@@ -80,14 +80,14 @@ def record(self, var1: T, var2: T) -> None:
         found1 = var1 in self._class_ids
         found2 = var2 in self._class_ids
 
-        if not found1 and not found2:
-            self._add(var1, var2)
-        elif found1:
+        if found1 and found2:  # both
+            self._merge(var1, var2)
+        elif found1:  # first only
             self._update(var1, var2)
-        elif found2:
+        elif found2:  # second only
             self._update(var2, var1)
-        else:
-            self._merge(var1, var2)
+        else:  # neither
+            self._add(var1, var2)
```

Comment:
Previously the else case (both true) would never be reached because either of the previous branches would trigger first.  cc @magnatelee

Reply:
good catch!